tests/unit/traits/is_iterator.cpp

```diff
@@ -0,0 +1,921 @@
+  {
+    using iterator = bidirectional_traversal_iterator;
+    HPX_TEST_MSG((is_output_iterator<iterator>::value),
+                 "bidirectional traversal input iterator");
+  }
+  {
+    using iterator = random_access_traversal_iterator;
+    HPX_TEST_MSG((is_output_iterator<iterator>::value),
+                 "random access traversal input iterator");
+  }
+}
+
+void is_input_iterator()
+{
+  using hpx::traits::is_input_iterator;
+
+  {
+    using iterator = std::ostream_iterator<int>;
+    HPX_TEST_MSG((not is_input_iterator<iterator>::value),
+                 "output iterator");
+  }
+  {
+    using iterator = std::istream_iterator<int>;
+    HPX_TEST_MSG((is_input_iterator<iterator>::value),
+                 "input iterator");
+  }
+  {
+    using iterator = typename std::forward_list<int>::iterator;
+    HPX_TEST_MSG((is_input_iterator<iterator>::value),
+                 "forward iterator");
+  }
+  {
+    using iterator = typename std::list<int>::iterator;
+    HPX_TEST_MSG((is_input_iterator<iterator>::value),
+                 "bidirectional iterator");
+  }
+  {
+    using iterator = typename std::vector<int>::iterator;
+    HPX_TEST_MSG((is_input_iterator<iterator>::value),
+                 "random access iterator");
+  }
+  {
+    using iterator = bidirectional_traversal_iterator;
+    HPX_TEST_MSG((is_input_iterator<iterator>::value),
+                 "bidirectional traversal input iterator");
+  }
+  {
+    using iterator = random_access_traversal_iterator;
+    HPX_TEST_MSG((is_input_iterator<iterator>::value),
+                 "random access traversal input iterator");
+  }
+}
+
+void is_forward_iterator()
+{
+  using hpx::traits::is_forward_iterator;
+
+  {
+    using iterator = std::ostream_iterator<int>;
+    HPX_TEST_MSG((not is_forward_iterator<iterator>::value), "output iterator");
+  }
+  {
+    using iterator = std::istream_iterator<int>;
+    HPX_TEST_MSG((not is_forward_iterator<iterator>::value), "input iterator");
+  }
+  {
+    using iterator = typename std::forward_list<int>::iterator;
+    HPX_TEST_MSG((is_forward_iterator<iterator>::value), "forward iterator");
+  }
+  {
+    using iterator = typename std::list<int>::iterator;
+    HPX_TEST_MSG((is_forward_iterator<iterator>::value),
+                 "bidirectional iterator");
+  }
+  {
+    using iterator = typename std::vector<int>::iterator;
+    HPX_TEST_MSG((is_forward_iterator<iterator>::value),
+                 "random access iterator");
+  }
+  {
+    using iterator = bidirectional_traversal_iterator;
+    HPX_TEST_MSG((is_forward_iterator<iterator>::value),
+                 "bidirectional traversal input iterator");
+  }
+  {
+    using iterator = random_access_traversal_iterator;
+    HPX_TEST_MSG((is_forward_iterator<iterator>::value),
+                 "random access traversal input iterator");
+  }
+  {
+    using iterator =
+      hpx::vector_iterator<int, std::vector<int>>;
+    HPX_TEST_MSG((is_forward_iterator<iterator>::value),
```

Comment:
If I'm reading the error message correctly, this test should replicate the error. However (at least locally), the test is passing.

Are you aware of any specialization of the `is_forward_iterator` template that might be influencing the larger test?

Reply:
I think I know what's going wrong. Let me see what I can do.

Reply:
As it turned out, I didn't know what went wrong ;-) Anyways, I was able to fix the issues. Those were caused by a problem in our `iterator_facade` implementation that simply missed to expose the post-increment operator. Sorry for the trouble.

Reply:
Update: post-increment was not missing, just broken. All should be fine now *fingers crossed*.